### PR TITLE
CS Questions  + Knowledge

### PR DESCRIPTION
**Pull request by RuiAAPeres**

### CODE REVIEW

File: questions.md [hunk 1]
```diff
@@ -132,5 +131,13 @@ Please keep in mind that this document is in a continuous state of flux, with ne
 - Have you heard of or used `Cocoapods`, `Carthage`, and the `Swift Package Manager`?
 - Which version control tools do you use? How much experience do you have with `git`? Can you explain what a `rebase` operation is?
 
+### Computer Science 101 (Questions + Knowledge)
+
+- What is a `deadlock`? How do you avoid it?
+- What is a `race condition`? How do you avoid it?
+- Knowledge about Big O notation
+- Knowledge about data structures: graphs, trees, queues, stacks, linked lists, hashtables and others. 
+- Knowledge about algorithms: sorting, searching and others.
```
Comment (diegopetrucci) [inline questions.md:140]: These sound a lot like whiteboard-y questions :S

What happens when you have a candidate that is not from a CS background and has little knowledge? Do we skip these or we reject the candidate?

How are these related to our work? (I'm referring mainly to the algorithm ones.)

Comment (RuiAAPeres) [reply questions.md:140]: Basic knowledge about this is a requirement. You don't need a CS background to know this stuff. 8h of study of these concepts is more than enough for most people.

**Review by wltrup — APPROVED** [review]

File: questions.md (hunk 1 above)
Comment (diegopetrucci) [reply questions.md:140]: Well I think it depends on the position. If we're aiming for seniors okay, for lower I think you can get far without them.

It all boils down to who (as in candidate) you are okay to filter out and possibly lose. If you are okay with that go ahead.
